feat(trace-tree-node): Consolidating profile attrs, transactionId and…

Abdkhan14 opened the pull request:
… attr assigment from parent 

Profiles: 
- Before: A set of (`TraceTree.Profile = {profile_id: string} | {profiler_id: string}`). Incredibly hard to manage.
- After: Added `profileId` and `profilerId` properties to the node. Added a `hasProfiles` getter too. 

TransactionId: 
- Currently we always look for and assign a parent node's transactionId if transactionId is undefined on a node. This is too generalized. 
- Changed the `BaseNode.transactionId` class to look for 'transaction_id' in `BaseNode.value`
- We look for parent transaction's id in `SpanNode` specifically 

Added a `findClosestTransaction` method that consolidates the use of the findEapParentTransaction + findNodestoreParentTransaction methods

Commit:
feat(trace-tree-node): Consolidating profile attrs, transactionId and attr assigment from parent

Commit:
feat(trace-tree-node): Fixing tests

Commit:
feat(trace-tree-node): Iterating

gggritso reviewed (approved):
Makes sense, bugbot notwithstanding

Abdkhan14 committed:
feat(trace-tree-node): Extending baseNode class methods to mitigate g… (#103282)

…uards usage

- Added `attributes` property to the baseNode class

- Added `resolveValueFromSearchKey` abstract method to the baseNode
class -> Used to map custom values to the searched string. For example:
If the searched string is `span.duration` we return `node.space[1]` for
spanNode types, that holds the node's duration.

- Added tests

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>
